crates/services/txpool/src/txpool.rs

```diff
@@ -76,6 +76,33 @@ pub struct TxPool<ViewProvider> {
     database: ViewProvider,
 }
 
+pub trait TxPoolGasPrice {
+    fn gas_price(&self, block_height: BlockHeight) -> Option<GasPrice>;
```

Comment:
Hmm, I think the port should only request either the price from the last block or for the next block. Overwise, it is harder to understand what your expectations are.

Reply:
That's what I started with. I definitely think that would be a cleaner interface, but the problem is I don't know what "current block" really means. Like every time you call it it gives a new gas price? or does it have some kind of tracking behind the scenes to know when a block was committed and now we're on to the next block?

I actually used this:
```pub struct GasPriceParams {
    block_height: BlockHeight,
}
```
instead of `BlockHeight` on the `ProducerGasPrice` and got lazy with the other ones. Honestly, I don't really know what the method should take yet, but `BlockHeight` seems like a good place to start, since it's absolute.